float64 posx
float64 posy
float64 posz

